src/crypto/sha256.cpp

```diff
@@ -465,22 +470,28 @@ bool SelfTest(TransformType tr) {
 
 TransformType Transform = sha256::Transform;
 TransformD64Type TransformD64 = sha256::TransformD64;
+TransformD64Type TransformD64_4way = nullptr;
 } // namespace
 
 std::string SHA256AutoDetect()
 {
+    std::string ret = "standard";
 #if defined(USE_ASM) && (defined(__x86_64__) || defined(__amd64__))
     uint32_t eax, ebx, ecx, edx;
     if (__get_cpuid(1, &eax, &ebx, &ecx, &edx) && (ecx >> 19) & 1) {
         Transform = sha256_sse4::Transform;
         TransformD64 = TransformD64Wrapper<sha256_sse4::Transform>;
-        assert(SelfTest(Transform));
-        return "sse4";
+#if defined(ENABLE_SSE41) && !defined(BUILD_BITCOIN_INTERNAL)
```

Comment:
Just because libbitcoinconsensus doesn't take advantage of 4way? Or some buildsystem limitation?

Reply:
Both.

I didn't feel like creating copies of all the architecture-specialized libs for use within libbitcoinconsensus - especially as it doesn't benefit it.